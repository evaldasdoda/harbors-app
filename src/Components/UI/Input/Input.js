import React from 'react';
import PropTypes from 'prop-types';

// Styles
require('./Input.scss');

const input = (props) => {
    return (
        <input
            data-testid="input"
            id={props.id}
            type={props.type}
            name={props.name}
            defaultValue={props.value}
            className={'INPUT form-control'}
            onChange={props.onChange}
            placeholder={props.placeholder}
            required={props.required}
            min={props.min}
        />
    );
};

input.propTypes = {
    type: PropTypes.string,
    id: PropTypes.string,
    name: PropTypes.string,
    value: PropTypes.string,
    onChange: PropTypes.func,
    placeholder: PropTypes.string,
    required: PropTypes.bool,
    min: PropTypes.string
};

export default input;
