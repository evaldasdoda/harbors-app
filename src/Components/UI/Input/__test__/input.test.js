import React from 'react';
import ReactDOM from 'react-dom';
import Input from '../Input';

import {render, cleanup} from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import renderer from 'react-test-renderer';

afterEach(cleanup)

it("renders without crashing", ()=>{
    const div = document.createElement("div");
    ReactDOM.render(<Input></Input>, div)
})

it("renders input correctly",()=>{
    const {getByTestId} = render(<Input type="date" name="testname" id="testid" value="2000-02-02" placeholder="test" required={true} min="2000-01-01" className="INPUT" onChange={jest.fn} />)
    expect(getByTestId('input'))
})


it("matches snapshot",()=>{
    const tree = renderer.create(
        <Input 
            type="date"
            name="testname"
            id="testid"
            defaultValue="2000-02-02"
            placeholder="test"
            required={true}
            min="2000-01-01"
            class="INPUT"
            onChange={jest.fn}
        />).toJSON();
    expect(tree).toMatchSnapshot();
})