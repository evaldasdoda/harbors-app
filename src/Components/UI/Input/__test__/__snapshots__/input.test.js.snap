// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<input
  className="INPUT form-control"
  data-testid="input"
  id="testid"
  min="2000-01-01"
  name="testname"
  onChange={[Function]}
  placeholder="test"
  required={true}
  type="date"
/>
`;
